### ISSUE TITLE
Warn about using `grep` outside conditionals when `set -e` is in effect

### ISSUE THREAD

**Issue by baumanj** [koalaman/shellcheck#1165]
#### For new checks and feature suggestions
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

Using grep in the context of `set -e`, it is not generally the intent to exit the script on non-match situations. When the `grep` is in the context of a conditional, this is avoided, but sometimes it's used to assign variables and the result is unexpected.

Real world example: [this line](https://github.com/habitat-sh/habitat/blob/b6bb5b491e8c1932f0efe43ee93decb1c37cb0e3/components/studio/bin/hab-studio.sh#L816) caused https://github.com/habitat-sh/habitat/issues/4834

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

set -e

matches=$(grep foo bar)
echo "${matches:-no matches}"
```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

```
In test.sh line 5:
matches=$(grep foo bar)
          ^-- SCXXXX: Consider 'grep ... || true' or else script will exit on no match due to set -e
```
